x-pack/plugins/rule_registry/server/event_log/log/event_query_builder.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getFlattenedObject } from '@kbn/std';
+import { estypes } from '@elastic/elasticsearch';
+import { esKuery } from '../../../../../../src/plugins/data/server';
+
+import { DeepPartial } from '../utils/utility_types';
+import { mergeFields } from '../utils/fields';
+import { EventLogParams } from './internal_api';
+import { IEventQueryBuilder, IEventQuery, SortingParams, PaginationParams } from './public_api';
+import { EventQuery } from './event_query';
+
+export class EventQueryBuilder<TEvent> implements IEventQueryBuilder<TEvent> {
+  private readonly params: EventLogParams;
+  private loggerName: string;
+  private fields: DeepPartial<TEvent> | null;
+  private kql: string;
+  private sorting: SortingParams;
+  private pagination: PaginationParams;
+
+  constructor(params: EventLogParams) {
+    this.params = params;
+    this.loggerName = '';
+    this.fields = null;
+    this.kql = '';
+    this.sorting = [{ '@timestamp': { order: 'desc' } }, { 'event.sequence': { order: 'desc' } }];
+    this.pagination = { page: 1, perPage: 20 };
+  }
+
+  public filterByLogger(loggerName: string): IEventQueryBuilder<TEvent> {
```

Comment:
I enjoy using these kind of builder pattern things, but it seems implementing it as part of this PR in this particular plugin adds a lot more than is needed here.

Reply:
I just decoupled this builder from `.search()`, so now, if you don't need it or would prefer not using it, you can call `.search()` directly without messing up with the builder:

```
export interface IEventLog<TEvent> extends IEventLoggerTemplate<TEvent> {
  getNames(): IndexNames;

  getQueryBuilder(): IEventQueryBuilder<TEvent>;

  search<TDocument = TEvent>(
    request: estypes.SearchRequest
  ): Promise<estypes.SearchResponse<TDocument>>;
}
```